src/App.tsx

```diff
@@ -15,36 +16,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  ALPHABET = 'Sort alphabetically',
+  LENGTH = 'Sort by length',
+  RESET = 'Reset',
+}
+
+const preparedGoods = [...goodsFromServer];
+
+const getSortedGoods = (order: SortType, isReversed: boolean): string[] => {
+  const sortedGoods = [...preparedGoods].sort((a, b) => {
+    switch (order) {
+      case SortType.ALPHABET:
+        return a.localeCompare(b);
+      case SortType.LENGTH:
+        return a.length - b.length;
+      default:
+        return 0;
+    }
+  });
+
+  return isReversed ? sortedGoods.reverse() : sortedGoods;
+};
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+  const [sortOrder, setSortOrder] = useState<SortType>(SortType.RESET);
+
+  const handleSortOrderChange = (order: SortType) => {
+    if (order === SortType.RESET) {
+      setIsReversed(false);
+    }
+
+    setSortOrder(order);
+  };
+
+  const handleReverseGoods = () => {
+    setIsReversed(prevState => !prevState);
+  };
```

Comment:
The function's names should start with a verb to clearly indicate what they do. Consider renaming `handleSortOrderChange` to `changeSortOrder` and `handleReverseGoods` to `toggleReverseGoods` to better reflect their actions.